shared/bash_remediation_functions/lineinfile.sh

```diff
@@ -0,0 +1,106 @@
+# Functions which minic Ansible's lineinfile module.
+# dependencies: die.sh, variables.sh
+
+function lineinfile() {
+    local path="$1"
+    local regex="$2"
+    local new_line="$3"
+    local create="$(normalize_boolean "$4" "false")"
+    local state="$(normalize_state "$5" "present")"
+    local insert_after="$6"
+    local insert_before="$7"
+    local insensitive="$(normalize_boolean "$8" "true")"
+
+    [ ! -e "$path" ] && [ "$create" != "true" ] && die "Path '$path' wasn't found on this system and not creating. Refusing to continue."
+    [ ! -e "$path" ] && [ "$create" == "true" ] && touch "$path"
+
+    if [ "$state" == "absent" ]; then
+        [ -n "$new_line" ] && echo "Ignoring parameter new_line in absent mode..."
+        echo "Saving $path to $path.bak and removing line defined by the regex '$regex'..."
+        if [ "$insensitive" == "true" ]; then
+            LC_ALL=C sed --in-place=.bak "/$regex/Id" "$path"
+        else
+            LC_ALL=C sed --in-place=.bak "/$regex/d" "$path"
```

Comment:
For discussion: Does it make sense to keep the `.bak` files? What is expected of it?
If multiple rules touch the same file, the `.bak` file contain all previous changes.